lib/nrf_modem_lib/lte_net_if/lte_net_if.c

```diff
@@ -301,29 +301,36 @@ static void pdn_event_handler(uint8_t cid, enum pdn_event event, int reason)
 
 static void lte_reg_handler(const struct lte_lc_evt *const evt)
 {
-	if (evt->type != LTE_LC_EVT_NW_REG_STATUS) {
-		return;
-	}
-
-	switch (evt->nw_reg_status) {
-	case LTE_LC_NW_REG_REGISTERED_HOME:
-		__fallthrough;
-	case LTE_LC_NW_REG_REGISTERED_ROAMING:
-		/* Mark serving cell as available. */
-		LOG_DBG("Registered to serving cell");
-		update_has_cell(true);
-		break;
-	case LTE_LC_NW_REG_SEARCHING:
-		/* Searching for a new cell, do not consider this cell loss unless it
-		 * fails (which will generate a new LTE_LC_EVT_NW_REG_STATUS event with
-		 * an unregistered status).
-		 */
-		break;
-	default:
-		LOG_DBG("Not registered to serving cell");
-		/* Mark the serving cell as lost. */
-		update_has_cell(false);
-		break;
+	if (evt->type == LTE_LC_EVT_MODEM_EVENT && evt->modem_evt == LTE_LC_MODEM_EVT_RESET_LOOP) {
+		LOG_WRN("The modem has detected a reset loop. LTE network attach is now "
+			"restricted for the next 30 minutes. Power-cycle the device to "
+			"circumvent this restriction or reset the modem to factory settings by "
+			"issuing the AT\%XFACTORYRESET AT command.");
```

Comment:
I'm not sure if this is something we want to state as a solution for a reset loop. It is good as a workaround for development, though I'm afraid someone might get the idea to add the factory reset to production code from this, causing a live device to continue spamming attach requests to the network to "eternity".

A suggestion is to keep the first sentence and remove the second.

Reply:
Fair point, I can remove the last part of the sentence.

Reply:
Removed